runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5154,37 +5154,37 @@ class INTERPRETER_CLASS
 	enterContinuationImpl(REGISTER_ARGS_LIST)
 	{
 		VM_BytecodeAction rc = EXECUTE_BYTECODE;
-		jboolean res = JNI_FALSE;
 
 		j9object_t continuationObject = *(j9object_t*)_sp;
 
 		buildInternalNativeStackFrame(REGISTER_ARGS);
 		updateVMStruct(REGISTER_ARGS);
 
-		res = enterContinuation(_currentThread, continuationObject);
+		if (enterContinuation(_currentThread, continuationObject)) {
+			_sendMethod = J9VMJDKINTERNALVMCONTINUATION_EXECUTE_METHOD(_currentThread->javaVM);
+			rc = GOTO_RUN_METHOD;
+		}
 
 		VMStructHasBeenUpdated(REGISTER_ARGS);
-		if (VM_VMHelpers::exceptionPending(_currentThread)) {
-			rc = GOTO_THROW_CURRENT_EXCEPTION;
-			goto done;
-		}
 
-		restoreInternalNativeStackFrame(REGISTER_ARGS);
```

Comment:
The restore is still necessary isn't it?

Reply:
no, since enterContinuation will not stack a new cInterpreter, so this call will directly return, the `yieldImpl(bool)` will be only place where we exit a Continuation and restores the internal native frame from carrier stack

Reply:
All of the switching is done in INLs, so the INL frame will be present and needs to be removed, I think.

Reply:
Ah, this is only for the initial entry into the continuation, not for switching.